test/typings/should_not_compile/index.ts

```diff
@@ -0,0 +1,26 @@
+import {createSelector} from '../../../src/reselect.d.ts';
+import * as common from '../common';
+
+
+const implicitlyTypedSelector = createSelector(
```

Comment:
Do you have any examples of negative tests written for type definitions? I think they could be very useful, but checking for _any_ compiler error in any test won't scale very well.

If there's no well-known solution, I'd rather leave these out to simplify integration with CI.